buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -276,6 +277,11 @@ public void stop(boolean tailLogs) {
         nodes.forEach(each -> each.stop(tailLogs));
     }
 
+    @Override
+    public void setNameCustomization(Function<String, String> nameCustomization) {
```

Comment:
This is a bit awkward due to the suffixing being done outside of this function. If we add this I think it might be better to simply defer name generation entirely to this function. We could rename this `setNodeNameGenerator()` which takes a `BiFunction<String, String, Integer>`. That way the passed function could get the cluster name and node index and generate the name rather than relying on `String.replace()`.

Reply:
that was my initial intention as well and I agree it would be cleaner, but that requires more changes as we don't store the ordinal of the node explicitly, it's just part of the node name, since I don't think there will be other users of this functionality it might be better to consolidate these tests not to rely on it rather than making more changes that will be specific to this one use-case.

Reply:
That sounds good. Is there an open issue to refactor these tests to remove the need for specific node names? If not can we create one?

Reply:
I'm not convinced we need an issue for this one, or small cleanups in general. 
The overhead if creating, maintain and prioritizing them is high and if you really need an issue to remember then the value provided might be questionable. The ones that are really worth doing you keep bumping into and there's no way to forget about them. I don't think there's anything worth documenting around this either.

Reply:
Your memory is clearly better than mine. If I don't write things down I forget. If creating issues is too burdensome then we need to look into that process. I don't think we should rely just on folks remembering to address this stuff.

Reply:
I didn't mean to say that I will personally remember them, what I mean to is that you won't need to, as you'll bump into it again and again. Unfortunately, if we were to have issues for everything we need to clean up in the  build right now it will simply be too many.

Reply:
> I didn't mean to say that I will personally remember them, what I mean to is that you won't need to, as you'll bump into it again and again.

Agreed. We tend to defer work not directly related to the specific PR we are working on though so these things (although we bump into them constantly) tend to get pushed aside indefinitely w/o some explicit goal to fix them.

> Unfortunately, if we were to have issues for everything we need to clean up in the build right now it will simply be too many.

I don't necessarily see a problem with that. This is software. There is always an infinite number of improvements to make 😄 

I guess my point is if we agree this should be fixed, we should make the explicit somewhere, and not rely on an individual to take it upon themselves to do the work at their arbitrary discretion. Maybe there is something lighter weight than GitHub issues that we can use to track this stuff?

Reply:
I think GH is fine for tracking these things, and I agree an issue is the right place. Even if it is low priority, having an issue is something someone working outside this area can find and pickup, or reference to increase the priority. Otherwise the discussions and decisions we've made in issues can be lost.

Reply:
@mark-vieira agreed with everything you said,I would add  do that that there is some time frame to it other than "eventually", unless we fear we run out of work and need this long list to pick from. The problem with long lists is that you then have to prioritize. 
@rjernst I  also agree with everything you said :) but in this  particular case, I don't see it as much interest to anything else and there isn't any discussion or decision to capture, I'm all for an issue if e.x. we decide how to fix it and want to document that taught process (even trough that could also become irrelevant in some cases). I understand the value of communication, but feel that some of these would just decrease signal to noise and do more harm than good. 

I sometimes take notes for the things I do want to get to in the foreseeable future on my GH project, I don't mind creating issues for those instead for better visibility.